databuilder/databuilder/extractor/openlineage_extractor.py

```diff
@@ -0,0 +1,93 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+from typing import Any, Dict, Iterator
+
+from pyhocon import ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.models.table_lineage import TableLineage
+
+LOGGER = logging.getLogger(__name__)
+
+
+class OpenLineageTableLineageExtractor(Extractor):
+    # Config keys
+    TABLE_LINEAGE_FILE_LOCATION = 'table_lineage_file_location'
+    CLUSTER_NAME = 'cluster_name'
+    OL_INPUTS_KEY = 'inputs_key'
+    OL_OUTPUTS_KEY = 'outputs_key'
+    OL_DATASET_NAMESPACE_KEY = 'namespace_key'
+    OL_DATASET_DATABASE_KEY = 'database_key'
+    OL_DATASET_NAME_KEY = 'dataset_name_key'
+    OL_DATASET_NAMESPACE_OVERRIDE = 'namespace_override'
+
+    """
+    An Extractor that creates Table Lineage between two tables based on OpenLineage event
+    """
+
+    def init(self, conf: ConfigTree) -> None:
+        """
+        :param conf:
+        """
+        self.conf = conf
+        self.table_lineage_file_location = conf.get_string(OpenLineageTableLineageExtractor.TABLE_LINEAGE_FILE_LOCATION)
+        self.cluster_name = conf.get_string(OpenLineageTableLineageExtractor.CLUSTER_NAME)
+        self.ol_inputs_key = conf.get_string(OpenLineageTableLineageExtractor.OL_INPUTS_KEY, default='inputs')
+        self.ol_outputs_key = conf.get_string(OpenLineageTableLineageExtractor.OL_OUTPUTS_KEY, default='outputs')
+        self.ol_namespace_key = conf.get_string(OpenLineageTableLineageExtractor.OL_DATASET_NAMESPACE_KEY, default='namespace')
+        self.ol_database_key = conf.get_string(OpenLineageTableLineageExtractor.OL_DATASET_DATABASE_KEY, default='database')
+        self.ol_dataset_name_key = conf.get_string(OpenLineageTableLineageExtractor.OL_DATASET_NAME_KEY, default='name')
+        self.ol_namespace_override = conf.get_string(OpenLineageTableLineageExtractor.OL_DATASET_NAMESPACE_OVERRIDE, default=None)
+        self._load_openlineage_event()
+
+    def _extract_dataset_info(self, openlineage_event: Any) -> Iterator[Dict]:
+        """
+        Yield input/output dict in form of amundsen table keys
+        """
+
+        for event in openlineage_event:
+            try:
+                in_and_outs = ((inputs, outputs)
+                               for inputs in event[self.ol_inputs_key]
+                               for outputs in event[self.ol_outputs_key])
+                for row in in_and_outs:
+                    yield {'input': self._amundsen_dataset_key(row[0]),
+                           'output': self._amundsen_dataset_key(row[1])}
+            except KeyError:
+                LOGGER.error(f'Cannot extract valid input or output from Openlineage event \n {event} ')
+
+    def _amundsen_dataset_key(self, dataset: Dict) -> str:
```

Comment:
we could describe what is the behavior of this method and that its default assumption is that it expects spark/hive table_name/path